react/Overlay.tsx

```diff
@@ -0,0 +1,75 @@
+import React, { useRef, useEffect, useState, FunctionComponent } from 'react'
+import Portal, { Props as PortalProps } from './components/Portal'
+
+interface Props extends PortalProps {
+  fullWindow: boolean
+}
+
+interface Position {
+  left: number
+  top: number
+}
+
+const Overlay: FunctionComponent<Props> = ({
+  children,
+  fullWindow,
+  target,
+}) => {
+  const container = useRef<HTMLDivElement>(null)
+  const [position, setPosition] = useState<Position>()
+
+  const updatePosition = () => {
+    if (!fullWindow && container.current) {
+      const bounds = container.current.getBoundingClientRect()
+      setPosition({
+        left: bounds.left,
+        top: bounds.top,
+      })
+    }
+  }
+
+  useEffect(updatePosition, [fullWindow])
```

Comment:
can you merge this effect with the one bellow? since this will run after the first render, I think you can just call updatePosition after adding the event listener.

you can also just define the `updatePosition` function directly inside the `useEffect`, since it will only be used there.